### PR TITLE
MM-47188 Migrate messageHtmlToComponent to TypeScript

### PR DESCRIPTION
**Pull request by Dismated**
#### Summary
Migrated file to TS

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-47188

#### Release Note
NONE

### CODE REVIEW

**Comment by sinansonmez** [comment]
Thanks for your contribution @Dismated Could you please run`make check-types` and `make check-style` in webapp. I see some type mismatches. After that I will have a second round of look

File: webapp/channels/src/utils/message_html_to_component.tsx [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    }
+
+    const processingInstructions: ProcessingInstructions[] = [
+
+        // Workaround to fix MM-14931
+        {
+            replaceChildren: false,
+            shouldProcessNode: (node) => node.type === 'tag' && node.name === 'input' && node.attribs.type === 'checkbox',
+            processNode: (node) => {
+                let newAttribs = JSON.parse(JSON.stringify(node.attribs));
```
Comment (sinansonmez) [inline webapp/channels/src/utils/message_html_to_component.tsx:67]: Compared to old version, I believe we are changing the behaviour here. If OK for you, we can keep the old version

File: webapp/channels/src/utils/message_html_to_component.tsx [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+                        <PostEditedIndicator
+                            postId={options.postId}
+                            editedAt={options.editedAt}
+                        />
+                    </React.Fragment>
+                ) : null;
+            },
+        },
+    ];
+    if (options) {
```
Comment (sinansonmez) [inline webapp/channels/src/utils/message_html_to_component.tsx:88]: Do we really need to introduce this if check?

**Review by sinansonmez — CHANGES_REQUESTED** [review]

**Comment by M-ZubairAhmed** [comment]
@Dismated do you need help with this PR

**Comment by M-ZubairAhmed** [comment]
@Dismated feel free to reopen the PR if you intend to work on it

**Pull request closed without merging**